KTB-07-study-mode

DenisGubernatorov opened the pull request:
1) Изменено правильное количество ответов для слов в словаре, для наглядности вывода вариантов ответа 2) learnedWords вынесено из when в область видимости основного цикла, для общего использования в when 3) Предварительный вариант вывода слов и вариантов для них. Выводит все неизученные слова, без ожидания пользователя ввода ответа. Реализовано в приватной функции

DenisGubernatorov committed:
KTB-06-statistic-mode
1) Изменено правильное количество ответов для слов в словаре, для наглядности вывода вариантов ответа
2) learnedWords вынесено из when в область видимости основного цикла, для общего использования в when
3) Предварительный вариант вывода слов и вариантов для них. Выводит все неизученные слова, без ожидания пользователя ввода ответа. Реализовано в приватной функции

DenisGubernatorov commented:
В задаче написано: " Там говорится о том, что для вариантов ответа должно браться максимальное количество невыученных слов (чтобы пользователь не угадал их методом исключения)." При таком варианте, если невыученных слов будет только 1 или 2. То и вариантов ответа, соответственно будет 1 или 2. При таком количестве это противоречит тому, что бы пользователь не смог угадать методом исключения

kaznikit commented:
> В задаче написано: " Там говорится о том, что для вариантов ответа должно браться максимальное количество невыученных слов (чтобы пользователь не угадал их методом исключения)." При таком варианте, если невыученных слов будет только 1 или 2. То и вариантов ответа, соответственно будет 1 или 2. При таком количестве это противоречит тому, что бы пользователь не смог угадать методом исключения

в таком варианте мы должны докидывать варианты ответа из списка уже выученных слов, чтобы у нас в сумме всегда было 4 варианта ответа

kaznikit commented:
> 1. Изменено правильное количество ответов для слов в словаре, для наглядности вывода вариантов ответа 2) learnedWords вынесено из when в область видимости основного цикла, для общего использования в when 3) Предварительный вариант вывода слов и вариантов для них. Выводит все неизученные слова, без ожидания пользователя ввода ответа. Реализовано в приватной функции

будь внимательнее к номерам задач. у тебя ПР называется KTB-6, ветка KTB-7, при этом код тоже для KTB-7)

kaznikit commented on src/main/kotlin/Main.kt:
```diff
@@ -45,3 +49,19 @@ fun main() {
     }
 }
+
+private fun studyStageIn(wordsToLearn: Set<Word>): Boolean {
```
функции нужно стараться называть в инфинитиве, тут больше подойдет `startLearning`

kaznikit commented on src/main/kotlin/Main.kt:
```diff
@@ -45,3 +49,19 @@ fun main() {
+        wordsToLearn.forEach {
+            println("Слово: ${it.original}")
+            println("Варианты ответа:")
+            val variants = wordsToLearn.take(wordsToLearn.size).shuffled()
```
нужно добавить к списку выученные слова, чтобы в сумме было 4 варианта ответа

kaznikit commented on src/main/kotlin/Main.kt:
```diff
@@ -45,3 +49,19 @@ fun main() {
 }
+
+private fun studyStageIn(wordsToLearn: Set<Word>): Boolean {
+    var isFinished = false
```
нам в целом ответ из функции не особо то и нужен) в случае когда мы выучили все слова юзер увидит об этом сообщение и сможет выбрать еще 1 пункт меню, например посмотреть статистику

DenisGubernatorov committed:
KTB-07-study-mode
1) Изменено название метода для запуска изучения
2) В функции startLearning убрано возвращаемое значение.
3) Доработка формирования списка ответов для слова. Если в списке для изучения меньше 4х слов, то недостающие варианты берутся из общего списка выученных слов
